src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/ProductPreferencesController.php

```diff
@@ -75,33 +60,23 @@ public function indexAction(Request $request)
             'help_link' => $this->generateSidebarLink($legacyController),
             'requireFilterStatus' => false,
             'form' => $form->createView(),
-        ];
+        ]);
     }
 
     /**
      * Process product preferences form.
      *
+     * @AdminSecurity("is_granted(['update', 'create','delete'], request.get('_legacy_controller'))",
+     *     message="You do not have permission to update this.",
```

Comment:
@colinegin should we update the wording to a generic one.
`You do not have permission to perform this action.`

Reply:
I agree with you @PierreRambaud, but let's ask to Louise, she's the master !

Reply:
2nd option too!

Reply:
Alright, thanks for the feedback. But if we choose this wording, we need to make sure that the user can get this permission otherwise we will induce wrong behavior.

Reply:
Well, it seems permissions can be set in Advanced Parameters > Team to manage the 'Product Settings' page of the back office. Let's pick the second option then: _You need permission to perform this action._ in _Admin.Notifications.Error_.

Reply:
 @matks Wording needs to be updated :)

Reply:
@PierreRambaud We're considering the idea of carrying out this in another PR after having decided new global rules in https://github.com/PrestaShop/PrestaShop/issues/12235